src/batched/KokkosBatched_SolveUTV_TeamVector_Internal.hpp

```diff
@@ -120,10 +120,10 @@ namespace KokkosBatched {
       value_type * W = w; /// m x nrhs
       const int ws0 = xs0 < xs1 ? 1 : nrhs, ws1 = xs0 < xs1 ? m : 1;
 
-      if (matrix_rank < m) {
+      if (matrix_rank < n) {
 	/// U is m x matrix_rank
 	/// T is matrix_rank x matrix_rank
-	/// V is matrix_rank m
+	/// V is matrix_rank x n
 	/// W = U^T B
 	TeamVectorGemmInternal<Algo::Gemm::Unblocked>
```

Comment:
No barriers or reductions at the end of Gemm, so team_barrier() aded since W is immediately reused in Trsm that follows.